lib/request_handler/validation/dry_backend.rb

```diff
@@ -0,0 +1,39 @@
+require_relative 'errors'
+require_relative 'backend'
+require_relative 'result'
+
+module RequestHandler
+  module Validation
+    class DryBackend < Backend
+      def self.valid_schema?(schema)
+        schema.respond_to?(:call)
+      end
+
+      def self.validate(value, schema, options: {})
+        result = if options.empty?
+                   schema.call(value)
+                 else
+                   schema.with(options).call(value)
+                 end
+
+        to_result(result)
+      rescue Dry::Types::ConstraintError => e
+        Result.new(output: nil, errors: { '' => e })
```

Comment:
hmm why's the key a empty string?

Reply:
Yeah i didnt know what else to use here since for a Dry::Types error there's is no key that is wrong. But in order to have a common output format with schema errors we need to return a hash. And then i thought an empty string is not that wrong since it indicates that there is no key... But I'm open for other ideas